py/sentry_streams/examples/billing.py

```diff
@@ -0,0 +1,108 @@
+import json
+from typing import Optional
+
+from sentry_streams.examples.billing_buffer import PendingBuffer
+from sentry_streams.pipeline import (
+    KafkaSource,
+    Map,
+    Pipeline,
+    Reduce,
+)
+from sentry_streams.user_functions.function_template import (
+    Accumulator,
+    AggregationBackend,
+)
+from sentry_streams.window import TumblingWindow
+
+Outcome = dict[str, str]
+
+
+def build_outcome(value: str) -> Outcome:
+
+    d: Outcome = json.loads(value)
+
+    return d
+
+
+class OutcomesBackend(AggregationBackend[PendingBuffer]):
```

Comment:
Let's say this was a redis storage. 
We would want to control the place we write the buffer to Redis. The product should only know about the abstract data structures they are writing like keys and values.
What about we had a KVBackend (Key Value pair Backend).
We extend it to provide a Redis implementation and make PendingBuffer provide a method that returns the keys and values for the storage.

You do not need to implement it here unless it is quick. But let's track this development.

Reply:
Couldn't `storage_map` act as placeholder for redis, or another key value store?

Reply:
Implemented, let me know what you all think

If we think it's not a good template as-is, I'll tackle in a separate PR